src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -154,15 +173,34 @@ public DashboardWebApplication()
         _app.MapGrpcService<OtlpMetricsService>();
         _app.MapGrpcService<OtlpTraceService>();
         _app.MapGrpcService<OtlpLogsService>();
+
+        static Func<IPEndPoint> CreateEndPointAccessor(ListenOptions options)
+        {
+            // We want to provide a way for someone to get the IP address of an endpoint.
+            // However, if a dynamic port is used, the port is not known until the server is started.
+            // Instead of returning the ListenOption's endpoint directly, we provide a func that returns the endpoint.
+            // The endpoint on ListenOptions is updated after binding, so accessing it via the func after the server
+            // has started returns the resolved port.
+            return () => options.IPEndPoint!;
```

Comment:
If it's stupid and it works, is it stupid? 🐴 

I did this instead of `IServerAddressesFeature` because there are multiple endpoints with different behaviors. The addresses feature doesn't provide enough information.